tests/js/core/widget/core/Drawer/Drawer.js

```diff
@@ -0,0 +1,684 @@
+/* global test, define, tau */
+(function () {
+	"use strict";
+	function runTests(Drawer, helpers) {
+
+		function initHTML() {
+			var HTML = helpers.loadHTMLFromFile("/base/tests/js/core/widget/core/Drawer/test-data/sample.html"),
+				parent = document.getElementById("qunit-fixture") || helpers.initFixture();
+
+			parent.innerHTML = HTML;
+		}
+
+		module("core/widget/core/Drawer", {
+			setup: initHTML
+		});
+
+		test("constructor", 2, function (assert) {
+			var widget = new Drawer();
+
+			assert.ok(!!widget, "instance of widget exists");
+			assert.ok(!!widget._ui, "cache for UI components exists");
+		});
+
+		test("Test of _build drawer", 5, function (assert) {
+			var widget = new Drawer(),
+				element = document.getElementById("drawer-1"),
+				result;
+
+			widget.options.overlay = true;
+
+			result = widget._build(element);
+
+			assert.equal(result, element, "method returns widget based element");
+			assert.ok(element.classList.contains("ui-drawer"), "widget class has been added to element");
+			assert.equal(element.style.top, "0px", "widget class has been added to element");
+			assert.ok(!!widget._ui.drawerOverlay, "overlay appended to widget");
+			assert.equal(widget._ui.drawerOverlay.style.visibility, "hidden", "overlay appended to widget");
+		});
+
+		test("Test of _init drawer", 4, function (assert) {
+			var widget = new Drawer(),
+				element = document.getElementById("drawer-1"),
+				result;
+
+			helpers.stub(widget, "_initLayout", function () {
+				assert.ok(true, "_initLayout has been called");
+			});
+
+			result = widget._init(element);
+
+			assert.equal(result, element, "method returns widget based element");
+			assert.equal(widget._ui.drawerPage, document.getElementById("page-1"), "page has been cached in widget");
+			assert.equal(widget._ui.drawerPage.style.overflowX, "hidden", "overflowX is hidden for page");
+		});
+
+		test("Test of handleEvent drawer", 34, function (assert) {
```

Comment:
Is correct count(34) for the 'test'?

Reply:
This is correct.
`widget.handleEvent(testEvent)` has been called 17 times and each request run 2 assertions.

Reply:
Okay!